services/ui-src/src/components/modals/AddEditEntityModal.tsx

```diff
@@ -85,7 +85,10 @@ export const AddEditEntityModal = ({
     } else {
       // create new entity
       dataToWrite.fieldData = {
-        [entityType]: [...currentEntities, { id: uuid(), ...filteredFormData }],
+        [entityType]: [
+          ...currentEntities,
+          { id: uuid(), type: entityType, ...filteredFormData },
```

Comment:
moved entityType to live within the entity itself, rather than being something a page is determining